Add endpoint to get situations status - ALEC-163

BenjaminJ opened the pull request:
Add endpoint to get situations status
https://issues.opennms.org/browse/ALEC-163

Situations cannot be accepted and rejected at the same time
https://issues.opennms.org/browse/ALEC-166

BenjaminJ committed:
Add endpoint to get situation status list

o implement getSituationsWithAlarmId in StaticSituationDatasource.java
o fix toSituationWithAlarmId in ApiMapper.java (status wasn't updated)
o implement getSituationsWithAlarmId in OpennmsDatasource.java
o retrieve situation status in OpennmsMapper.java
o feature/ui cleanup
o fix package name
o rename situation refused to situation rejected
o create new interface for situation rest
o fix situations shouldn't be REJECTED and ACCEPTED at the same time SituationRestImpl.java (ALEC-166)

ALEC-163, ALEC-166

BenjaminJ committed:
Add endpoint to get situation status list

o fix package name
o rename situation refused to situation rejected
o create new interface for situation rest

ALEC-163

BenjaminJ committed:
Merge remote-tracking branch 'origin/feature/ALEC-163' into feature/ALEC-163

# Conflicts:
#	features/ui/src/main/java/org/opennms/alec/data/SituationStatus.java
#	features/ui/src/main/java/org/opennms/alec/data/SituationStatusImpl.java
#	features/ui/src/main/java/org/opennms/alec/jackson/JacksonAlarm.java
#	features/ui/src/main/java/org/opennms/alec/jackson/JacksonSituation.java
#	features/ui/src/main/java/org/opennms/alec/rest/SituationRest.java
#	features/ui/src/main/java/org/opennms/alec/rest/SituationRestImpl.java
#	features/ui/src/test/java/org/opennms/alec/data/JacksonSituationTest.java

RangerRick commented on features/ui/src/test/java/org/opennms/alec/rest/ALECRestImplTest.java:
```diff
@@ -38,12 +44,6 @@
-import org.opennms.alec.jackson.Configuration;
-import org.opennms.alec.jackson.EngineParameter;
-import org.opennms.alec.jackson.EngineParameterImpl;
-import org.opennms.alec.jackson.KeyEnum;
```
someday we'll make IDE templates for everyone so we stop playing the import dance :D

(not your fault, just complaining)

RangerRick reviewed (dismissed):
This all looks straightforward to me.

Just looking at the diff made the changes to the OpennmsMapper look a bit odd, but seeing it in context it all makes sense.

BenjaminJ committed:
Fix sonar cloud quality gate

o Either re-interrupt this method or rethrow the "InterruptedException" that can be caught here. OpennmsDatasource.java
o Update this method so that its implementation is not identical to "getSituations" on line 46. StaticSituationDatasource.java

BenjaminJ committed:
Fix sonar cloud quality gate

o Either re-interrupt this method or rethrow the "InterruptedException" that can be caught here. OpennmsDatasource.java

BenjaminJ committed:
Fix sonar cloud quality gate - "InterruptedException" should not be ignored

cgorantla reviewed (approved):
LGTM!
